gossip/c_block_callbacks.go

```diff
@@ -178,7 +178,17 @@ func consensusCallbackBeginBlockFn(
 					}
 				}
 
-				evmProcessor := blockProc.EVMModule.Start(blockCtx, statedb, evmStateReader, onNewLogAll, es.Rules, es.Rules.EvmChainConfig(store.GetUpgradeHeights()))
+				prevRandao := inter.ComputePrevRandao(confirmedEvents)
```

Comment:
You could move the initialization of the `block` up here and read the prevRandao value from the block. This eliminated the need for the public `ComputePrevRandao`, the `SetPrevRandao` on the struct, and would define a canonical source for this information.
```suggestion
				var block = &inter.Block{
					Time:    blockCtx.Time,
					Atropos: cBlock.Atropos,
					Events:  hash.Events(confirmedEvents),
				}
				prevRandao := block.GetPrevRandao()
```